src/pages/MainQuizScreen.tsx

```diff
@@ -0,0 +1,87 @@
+import { useState } from 'react';
+import MyButton from '../components/UI/button/MyButton';
+import ProgressBar from '../components/UI/progress_bar/ProgressBar';
+import QuestionBlock from '../components/UI/question_block/QuestionBlock';
+import MySelect from '../components/UI/select/MySelect';
+import Timer from '../components/UI/timer/Timer';
+
+const handleButtonClick = () => {
+  console.log('Button clicked!');
+};
+
+const quizQuestions = [
+  { id: 1, content: 'The Earth is flat.' },
+  { id: 2, content: 'Which planet is known as the Red Planet?' },
+  { id: 3, content: 'Humans share 50% of their DNA with bananas.' },
+  { id: 4, content: 'What is the capital of France?' },
+  { id: 5, content: 'The chemical symbol for gold is Au.' },
+];
+
+const answers = [
+  { id: 1, questionId: 1, option: 'True' },
+  { id: 2, questionId: 1, option: 'False' },
+  { id: 3, questionId: 2, option: 'Earth' },
+  { id: 4, questionId: 2, option: 'Mars' },
+  { id: 5, questionId: 3, option: 'True' },
+  { id: 6, questionId: 3, option: 'False' },
+  { id: 7, questionId: 4, option: 'Berlin' },
+  { id: 8, questionId: 4, option: 'Madrid' },
+  { id: 9, questionId: 4, option: 'Paris' },
+  { id: 10, questionId: 4, option: 'Lisbon' },
+  { id: 11, questionId: 5, option: 'True' },
+  { id: 12, questionId: 5, option: 'False' },
+];
+
+const correctAnswers = [
+  { answerId: 2, questionId: 1 },
+  { answerId: 4, questionId: 2 },
+  { answerId: 5, questionId: 3 },
+  { answerId: 9, questionId: 4 },
+  { answerId: 11, questionId: 5 },
+];
+
+const getAnswersByQuestionId = (questionId: number) => {
+  answers.filter((answer) => answer.questionId === questionId);
+};
+
+const isCorrectAnswer = (questionId: number, chosenAnswerId: number) => {
+  return Boolean(
+    correctAnswers.find(
+      (answer) =>
+        answer.questionId === questionId && answer.answerId === chosenAnswerId,
+    ),
+  );
+};
+
+const MainQuizScreen: React.FC = () => {
+  const [currentQuestionIndex, setCurrentQuestionIndex] = useState(0);
+
+  const nextQuestion = () => {
+    setCurrentQuestionIndex((prevIndex) => prevIndex + 1);
+  }; // for the future implementation of changing questions;
+
+  return (
+    <>
+      {currentQuestionIndex < quizQuestions.length ? (
+        <QuestionBlock
+          questionText={quizQuestions[currentQuestionIndex].content}
+        />
+      ) : (
+        <div className="quizCompletion">Quiz completed!</div>
+      )}
```

Comment:
I think then refactoring the structure doesn't make sense since we don't use answers, utilities, and so on
so, first of all:
```tsx
const [currentQuestionId, setCurrentQuestionId] = useState<number | undefined>(1);

const [question, answers] = useMemo(() => {
  const _question = quizQuestions.find(q => q.id === currentQuestionId);
  const _answers = getAnswersByQuestionId(currentQuestionId);
  return [_question, _answers]
}, [currentQuestionId]);

const handleNextQuestion = () => {
  const nextQuestion = quizQuestions.find(q => q.id === currentQuestionId + 1);
  setCurrentQuestionId(nextQuestion?.id)
}

//...
return (
  <>
    {question ? 
      <QuestionBlock questionText={question.content} /> : 
      <div className="quizCompletion">Quiz completed!</div>
    }
    // the rest of UI
  </>
)
```